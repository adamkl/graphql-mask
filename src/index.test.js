// @flow

const { parse, buildASTSchema } = require("graphql");

const graphqlMask = require("../src");

test("removing fields from query that aren't in schema", () => {
  const astSchema = buildASTSchema(
    parse(`
    type Query {
      this: [Bar]!
      random: String
      thing: Int!
    }
    type Bar {
      id: ID!
      is: String
      a: Int
    }
    `)
  );

  const resultQueryString = graphqlMask(
    astSchema,
    `
    query RandomQuery {
      this {
        is
        a
      }
      random
      thing
      notInSchema
    }
  `
  );
  expect(resultQueryString).toMatchSnapshot();
});

test("removing arguments that don't exist in schema", () => {
  const astSchema = buildASTSchema(
    parse(`
    type Query {
      something(bar: Int): String
    }
    `)
  );

  const resultQueryString = graphqlMask(
    astSchema,
    `
    query RandomQuery {
      something(foo: "foo", bar: 4)
    }
  `
  );
  expect(resultQueryString).toMatchSnapshot();
});

test("passing a string schema instead of a parsed one", () => {
  const resultQueryString = graphqlMask(
    `
    type Query {
      two: Int
    }`,
    `
    query RandomQuery {
      one
      two      
    }
  `
  );
  expect(resultQueryString).toMatchSnapshot();
});
