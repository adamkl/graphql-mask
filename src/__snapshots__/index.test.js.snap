// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passing a string schema instead of a parsed one 1`] = `
"query RandomQuery {
  two
}
"
`;

exports[`removing arguments that don't exist in schema 1`] = `
"query RandomQuery {
  something(bar: 4)
}
"
`;

exports[`removing fields from query that aren't in schema 1`] = `
"query RandomQuery {
  this {
    is
    a
  }
  random
  thing
}
"
`;
